### PR TITLE
[RFC] vim-patch:7.4.{984,1093}

### PR DESCRIPTION
**Pull request by watiko**
#### vim-patch:7.4.984

Problem:    searchpos() always starts searching in the first column, which is
            not what some people expect. (Brett Stahlman)
Solution:   Add the 'z' flag: start at the specified column.

https://github.com/vim/vim/commit/ad4d8a192abf44b89371af87d70b971cd654b799
#### vim-patch:7.4.1093

Problem:    Typo in test goes unnoticed.
Solution:   Fix the typo. Give error for wrong arguments to cursor().
            (partly by Hirohito Higashi) Add a test for cursor().

https://github.com/vim/vim/commit/24c4d539eed33e8073f8f9fe2bee497bbba935a4

### CODE REVIEW

File: src/nvim/search.c [hunk 1]
```diff
@@ -453,25 +453,24 @@ void last_pat_prog(regmmatch_T *regmatch)
...
+// if (options & SEARCH_END) return position at end of match
+// if (options & SEARCH_START) accept match at pos itself
+// if (options & SEARCH_KEEP) keep previous search pattern
+// if (options & SEARCH_FOLD) match only once in a closed fold
+// if (options & SEARCH_PEEK) check for typed char, cancel search
+// if (options & SEARCH_COL) start at pos->col instead of zero
+//
+// Return FAIL (zero) for failure, non-zero for success.
+// Returns the index of the first matching
+// subpattern plus one; one if there was none.
```
Comment (cacplate) [inline src/nvim/search.c]: This could be doxygen style since you already change it.

File: runtime/doc/eval.txt [hunk 2]
```diff
@@ -5427,14 +5427,15 @@ search({pattern} [, {flags} [, {stopline} [, {timeout}]]])	*search()*
...
 		'n'	do Not move the cursor
-		'p'	return number of matching sub-pattern (see below)
-		's'	set the ' mark at the previous location of the cursor
-		'w'	wrap around the end of the file
-		'W'	don't wrap around the end of the file
+		'p'	return number of matching sub-Pattern (see below)
+		's'	Set the ' mark at the previous location of the cursor
+		'w'	Wrap around the end of the file
+		'W'	don't Wrap around the end of the file
+		'z'	start searching at the cursor column instead of zero
```
Comment (oni-link) [inline runtime/doc/eval.txt]: Zero instead of zero?

File: src/nvim/search.c [hunk 3]
```diff
@@ -571,16 +570,15 @@ int searchit(
         if (tm != NULL && profile_passed_limit(*tm))
           break;
 
-        /*
-         * Look for a match somewhere in line "lnum".
-         */
+        // Look for a match somewhere in line "lnum".
+        colnr_T col = at_first_line && (options & SEARCH_COL)
+                      ? pos->col : (colnr_T)0;
```
Comment (oni-link) [inline src/nvim/search.c]: `(colnr_T) 0` -> `0`, also fits in one line.

File: src/nvim/version.c [hunk 4]
```diff
@@ -197,7 +197,7 @@ static int included_patches[] = {
   // 1096,
   // 1095 NA
   // 1094,
-  // 1093,
+  // 1093 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Why is this NA? `f_cursor()`in eval.c is also modified (error message, function comment).

File: runtime/doc/eval.txt (hunk 2 above)
Comment (watiko) [inline runtime/doc/eval.txt]: I don't think so. `Zero` should be used the first word of sentence.

File: src/nvim/version.c (hunk 4 above)
Comment (watiko) [inline src/nvim/version.c]: Oh, I missed it...

File: runtime/doc/eval.txt (hunk 2 above)
Comment (oni-link) [inline runtime/doc/eval.txt]: The other changes here do the same to help with the meaning of the flag character.
`'b'` -> Backward
`'c'` -> Cursor
...
`'z'` -> Zero

Comment (watiko) [inline runtime/doc/eval.txt]: Ah, I see. I will do it.

BTW, will you send it to vim_dev?

**Comment by watiko** [comment]
Updated. Thanks: @cacplate @oni-link

**Pull request merged**